ambry-store/src/main/java/com.github.ambry.store/IndexValue.java

```diff
@@ -318,28 +378,30 @@ ByteBuffer getBytes() {
             : (int) operationTimeInMs);
         value.putShort(accountId);
         value.putShort(containerId);
+        value.putShort(updateVersion);
         value.position(0);
         break;
       default:
-        throw new IllegalArgumentException("Unsupported version " + version + " for IndexValue ");
+        throw new IllegalArgumentException("Unsupported formatVersion " + formatVersion + " for IndexValue ");
     }
     return value;
   }
 
   @Override
   public String toString() {
     return "Offset: " + offset + ", Size: " + getSize() + ", Deleted: " + isFlagSet(Flags.Delete_Index)
-        + ", TTL Updated: " + isFlagSet(Flags.Ttl_Update_Index) + ", ExpiresAtMs: " + getExpiresAtMs()
-        + ", Original Message Offset: " + getOriginalMessageOffset() + (version != PersistentIndex.VERSION_0 ? (
-        ", OperationTimeAtSecs " + getOperationTimeInMs() + ", AccountId " + getAccountId() + ", ContainerId "
-            + getContainerId()) : "");
+        + ", TTL Updated: " + isFlagSet(Flags.Ttl_Update_Index) + ", Undelete Updated: " + isFlagSet(
+        Flags.Undelete_Index) + ", ExpiresAtMs: " + getExpiresAtMs() + ", Original Message Offset: "
+        + getOriginalMessageOffset() + (formatVersion != PersistentIndex.VERSION_0 ? (", OperationTimeAtSecs "
```

Comment:
Looks like we need to compare / look for versions 1 and 2 as well to distinguish from 3?